agent/engine/docker_task_engine.go

```diff
@@ -958,12 +965,38 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 		return metadata
 	}
 	pullSucceeded := metadata.Error == nil
-	if pullSucceeded {
+	findCachedImage := false
+	if !pullSucceeded {
+		// If Agent failed to pull an image when
+		// 1. DependentContainersPullUpfront is enabled
+		// 2. ImagePullBehavior is not set to always
+		// search the image in local cached images
+		if engine.cfg.DependentContainersPullUpfront.Enabled() && engine.cfg.ImagePullBehavior != config.ImagePullAlwaysBehavior {
+			if _, err := engine.client.InspectImage(container.Image); err != nil {
+				seelog.Errorf("Task engine [%s]: failed to find cached image %s for container %s",
+					task.Arn, container.Image, container.Name)
+				// Stop the task if the container is an essential container,
+				// and the image is not available in both remote and local caches
+				if container.IsEssential() {
+					task.SetKnownStatus(apitaskstatus.TaskStopped)
```

Comment:
I'd only change the desired status here as we don't know the task is stopped at this point.

Reply:
Removed `task.SetKnownStatus(apitaskstatus.TaskStopped)`, and keep  `task.SetDesiredStatus(apitaskstatus.TaskStopped)`. Thank you!